server/game/cards/02_SHD/leaders/BobaFettDaimyo.ts

```diff
@@ -0,0 +1,43 @@
+import AbilityHelper from '../../../AbilityHelper';
+import { LeaderUnitCard } from '../../../core/card/LeaderUnitCard';
+import { RelativePlayer, WildcardCardType } from '../../../core/Constants';
+
+export default class BobaFettDaimyo extends LeaderUnitCard {
+    protected override getImplementationId() {
+        return {
+            id: '9334480612',
+            internalName: 'boba-fett#daimyo',
+        };
+    }
+
+    protected override setupLeaderSideAbilities() {
+        this.addTriggeredAbility({
+            title: 'When you play a unit that has 1 or more keywords: You may exhaust this leader. If you do, give a friendly unit +1/+0 for this phase.',
+            cost: AbilityHelper.costs.exhaustSelf(),
+            optional: true,
+            when: {
+                onCardPlayed: (event, context) =>
+                    event.card.isUnit() && event.card.controller === context.source.controller && event.card.keywords.length > 0
```

Comment:
We might have a better way to check if a card has a keyword, but I'm not 100% sure. Veld can say for sure

Reply:
for a particular keyword we have method, but for any keyword I think not

Reply:
Yeah pretty sure we don't, but this comes up very infrequently so I don't feel the need to add a bespoke method for it